Gems/Meshlets/Code/Source/Meshlets/MeshletsAssets.cpp

```diff
@@ -0,0 +1,547 @@
+                float ty = ((vtxIndex / 5) % 5) * 0.25f;
+                mesh.vertices[vtxIndex].tx = tx;
+                mesh.vertices[vtxIndex].ty = ty;
+            }
+        }
+
+        Data::Instance<RPI::ShaderResourceGroup> MeshletsModel::CreateShaderResourceGroup(
+            Data::Instance<RPI::Shader> shader,
+            const char* shaderResourceGroupId,
+            [[maybe_unused]] const char* moduleName)
+        {
+            Data::Instance<RPI::ShaderResourceGroup> srg = RPI::ShaderResourceGroup::Create(shader->GetAsset(), AZ::Name{ shaderResourceGroupId });
+            if (!srg)
+            {
+                AZ_Error(moduleName, false, "Failed to create shader resource group");
+                return nullptr;
+            }
+            return srg;
+        }
+
+        uint32_t MeshletsModel::CreateMeshlets(GeneratorMesh& mesh)
+        {
+            const size_t max_vertices = 64;
+            const size_t max_triangles = 124; // NVidia-recommended 126, rounded down to a multiple of 4
+            const float cone_weight = 0.5f;   // note: should be set to 0 unless cone culling is used at runtime!
+
+            //----------------------------------------------------------------
+            size_t max_meshlets = meshopt_buildMeshletsBound(mesh.indices.size(), max_vertices, max_triangles);
+
+/*
+            // NO scan seems to return more localized meshlets
+            m_meshletsData.meshlets.resize(meshopt_buildMeshlets(
+            &m_meshletsData.meshlets[0], &m_meshletsData.meshlet_vertices[0], &m_meshletsData.meshlet_triangles[0],
+            &mesh.indices[0], mesh.indices.size(),
+            &mesh.vertices[0].px, mesh.vertices.size(), sizeof(GeneratorVertex),
+            max_vertices, max_triangles, cone_weight));
+            
+            if (!m_meshletsData.meshlets.size())
+            {
+                printf("Error generating meshlets - no meshlets were built\n");
+                return 0;
+            }
+
+            // this is an example of how to trim the vertex/triangle arrays when copying data out to GPU storage
+            const meshopt_Meshlet& last = m_meshletsData.meshlets.back();
+            m_meshletsData.meshlet_vertices.resize(last.vertex_offset + last.vertex_count);
+            m_meshletsData.meshlet_triangles.resize(last.triangle_offset + ((last.triangle_count * 3 + 3) & ~3));
+            uint32_t meshletsAmount = (uint32_t) m_meshletsData.meshlets.size();
+            //----------------------------------------------------------------
+            */
+
+            ////////////////////////////
+            std::vector<meshopt_Meshlet> meshlets(max_meshlets);
+            std::vector<unsigned int> meshlet_vertices(max_meshlets * max_vertices);        // Vertex Index indirection map
+            std::vector<unsigned char> meshlet_triangles(max_meshlets * max_triangles * 3); // Meshlet triangles into the vertex index indirection - local to meshlet.
```

Comment:
Should these use `AZStd::vector` instead?  There are a few other places `std::vector` is used.

Reply:
guys, such data conversions between two different libraries is never guaranteed and even if will work now, might fail in the future.  When connecting to external library you can either convert all data and that might be time consuming, or keep minimal required structures as the interface of data between the two.  The latter was chosen for this - both are correct. Assuming data compatibility is a very risky assumption that might be correct now but will not hold water in the future.